### PR TITLE
Added Cheat Sheet

### LINKED ISSUE #591: Documentation: printable cheat sheet including selected recipes
Some libraries provide 'cheat sheets' i.e. summary reference pages, designed to be online or on-paper or both, summarising important parts of their library's API and/or functionality and/or outputs. They may be listed under other names than 'cheat sheet', but the principle is the same, e.g. ''10 Minutes to X' (or 'N minutes to X') pages are quite similar in scope.

It would be nice if we can have one of more cheat sheets for cf-python. They could include summaries of our core functionalities, some notable recipes, and , etc.

Ideally this would be available in a printable format, suitable for A3 or A4 size printing, so anyone old school or otherwise can print one out to refer to off-screen. It could be divided into beginner, intermediate and advanced sections or sheets also.

Nice examples I have seen recently as inspiration are:

* [matplotlib](https://matplotlib.org/cheatsheets/);
* [pandas](https://pandas.pydata.org/Pandas_Cheat_Sheet.pdf), linked from [this page](https://pandas.pydata.org/docs/getting_started/index.html?highlight=cheat#tutorials);
* [dask](https://docs.dask.org/en/stable/10-minutes-to-dask.html).

***

This is something else you might want to work on at some point, @bewithankit!

### PR DESCRIPTION
**Pull request by bewithankit**
Resolves #591.

### CODE REVIEW

File: docs/source/cheat_sheet.rst [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+|                          | *Aggregate allowing for missing coordinate identities to be inferred:*                                                                                               |
+|                          |                                                                                                                                                                      |
+|                          | .. code-block:: python                                                                                                                                               |
+|                          |                                                                                                                                                                      |
+|                          |       >>> f = cf.read("~/realization/PRMSL.1941_mem*.nc")[0:50]                                                                                                      |
+|                          |       >>> ensemble = cf.aggregate(f, dimension=("realization",), relaxed_identities=True)[0]                                                                         |
```
Comment (davidhassell) [inline docs/source/cheat_sheet.rst:301]:
```suggestion
```
As discussed offline, we'll remove this for now, and decide how best to represent aggregation later.

File: docs/source/cheat_sheet.rst [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+cf-python.
+
+.. contents::
+   :local:
+   :backlinks: entry
+
++--------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+|         Operation        |                                                                                How to?                                                                               |
++==========================+======================================================================================================================================================================+
+|        **Install**       | *Install cf-python as root, with any missing dependencies, and the cf-plot visualisation package using pip:*                                                         |
```
Comment (davidhassell) [inline docs/source/cheat_sheet.rst:24]:
```suggestion
|        **Install**       | *Install cf-python, including any required dependencies, and the cf-plot visualisation package using pip:*                                                         |
```

File: docs/source/cheat_sheet.rst [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+|                          | *Temporal maxima:*                                                                                                                                                   |
+|                          |                                                                                                                                                                      |
+|                          | .. code-block:: python                                                                                                                                               |
+|                          |                                                                                                                                                                      |
+|                          |    >>> max = f.collapse('maximum', axes='T')                                                                                                                         |
+|                          |                                                                                                                                                                      |
```
Comment (davidhassell) [inline docs/source/cheat_sheet.rst:145]:
```suggestion
```

**Review by davidhassell — CHANGES_REQUESTED** [review]
Looks great. Some suggestions ...

**Comment by sadielbartholomew** [comment]
I'm aware this is still in draft form, but from a quick peek it is looking fantastic. If you'd like a second review (though probably not necessary) feel free to assign me when it is ready.

**Comment by bewithankit** [comment]
Thanks for the suggestions David! 
Sadie, I'll add a few things and request a review from you!

**Comment by bewithankit** [comment]
Hi @sadielbartholomew, I have marked it ready for review. You could either generate the documentation on your system or just head to https://github.com/bewithankit/cf-python/blob/cheatsheet/docs/source/cheat_sheet.rst where it renders properly!

**Review by sadielbartholomew — APPROVED** [review]
I've raised several minor comments, but overall this is looking excellent. Let me know what you think, but this close to merge IMO...

**Comment by sadielbartholomew** [comment]
Thanks for addressing the feedback, Ankit. @davidhassell are you happy with the present state, in which case I think we can merge?

**Comment by bewithankit** [comment]
Hi Sadie, thanks again for the suggestions. As the Cheat Sheet will be an addition to the sidebar menu, all the pages will need to be regenerated again. I think David said we'll add it in the next release as it'll be easier that way!

**Comment by sadielbartholomew** [comment]
Sounds good. I'm looking forward to seeing it in the documentation. Thanks Ankit.

File: docs/source/cheat_sheet.rst [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+|                                                                                  |       >>> jan_2023 = temp.subspace(T=cf.year(2023) & cf.month(1))                                                                                                               |
+|                                                                                  |       >>> annual_avg_61_90 = annual_global_avg.subspace(T=cf.year(cf.wi(1961, 1990)))                                                                                           |
++----------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+|                             **Accessing constructs**                             | *Select constructs by identity:*                                                                                                                                                |
```
[the remark this answers is not in the document]
Comment (bewithankit) [reply docs/source/cheat_sheet.rst:224]: Thanks for the suggestion David. I thought users might not know all the metadata constructs so thought of putting the `constructs` container! I have done the changes, let me know if those are correct, thanks!

File: docs/source/cheat_sheet.rst [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+|                                                                                  |       >>> d = f.constructs.filter_by_ncvar('time')                                                                                                                              |
+|                                                                                  |       >>> d = f.constructs.filter_by_ncvar('time', 'lat')                                                                                                                       |
++----------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+|                                  **Regridding**                                  | **Spherical regridding for regridding data between domains with spherical coordinate systems:**                                                                                 |
```
[the remark this answers is not in the document]
Comment (bewithankit) [reply docs/source/cheat_sheet.rst:265]: Hi David, I didn't get this!

[the remark this answers is not in the document]
Comment (davidhassell) [reply docs/source/cheat_sheet.rst:265]: Hi ANkit, sorry, it did get a bit confusing! I'm suggesting replacing `**Spherical regridding for regridding data between domains with spherical coordinate systems:**` with 
```
**Regridding in spherical polar coordinates**
```

File: docs/source/cheat_sheet.rst [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+|                                                                                  |         >>> fl = cf.read('~/file*.nc', select='air_temperature')                                                                                                                |
+|                                                                                  |         >>> fl = cf.read('~/cf-python/docs/source/sample_files/file*.nc', select='ncvar%q')                                                                                     |
+|                                                                                  +---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+|                                                                                  | **Selecting field:**                                                                                                                                                            |
```
Comment (davidhassell) [inline docs/source/cheat_sheet.rst:78]:
```suggestion
|                                                                                  | **Selecting fields:**                                                                                                                                                            |
```

**Review by davidhassell — APPROVED** [review]
Great. One minor suggestion - merge at will.

**Comment by bewithankit** [comment]
Hi David, just wanted to confirm if I should merge it now or do it in the next release as we had talked?

**Comment by davidhassell** [comment]
Hi - I think that you can merge it now, it just won't show up till the next release when we rebuild the docs (well the it will appear in the side bar of index.html, but won't work - that's OK for a few days)

**Pull request merged**